### PR TITLE
Implemented the infrastructure needed to incorporate AFL fuzzer easily

### PR DESCRIPTION
**Pull request by SharonBrizinov**

### CODE REVIEW

File: bin/posix/setup_posix.sh [hunk 1]
```diff
@@ -1,2 +1,2 @@
-cmake -DOpENer_PLATFORM:STRING="POSIX" -DCMAKE_BUILD_TYPE:STRING="" -DBUILD_SHARED_LIBS:BOOL=OFF ../../source
+cmake -DCMAKE_C_COMPILER=gcc -DOpENer_PLATFORM:STRING="POSIX" -DCMAKE_BUILD_TYPE:STRING="" -DBUILD_SHARED_LIBS:BOOL=OFF ../../source
```
Comment (CapXilinx) [inline bin/posix/setup_posix.sh:1]: Why is this change needed?

Comment (SharonBrizinov) [reply bin/posix/setup_posix.sh:1]: Technically it's not needed. However, in my local setup when I first ran the AFL setup it caused cmake to remember the last compiler I used and so when I later wanted to compile the release version it used the AFL compiler by default.

To eliminate any potential confusion I specified explicitly the compiler I want to use
